### DISCUSSION TITLE
write atom on async atom?

### DISCUSSION THREAD

**Discussion by prrrcl** [pmndrs/jotai#585] [Q&A] [answered]
hi!

If i write this: 

````
export const DATA = atom(
  async () => {
    // ...fetch data and return json
  },
  (get, set, update) => {
    set(DATA, (previous) => ({ ...previous, ...update }));
  }
);

````
I got: `Error: no atom init`

What is the correct way for update DATA atom?

**Comment by dai-shi** [comment, marked as answer]
It's prohibited to write a value directly to a derived atom. (because it may cause inconsistencies.)

This pattern is implemented as `atomWithDefault` in `jotai/utils`. You could also do the similar way on your end.

**Reply by prrrcl** [reply]
Oh, I didn’t know about the inconsistencies on that. 

Thanks a lot @dai-shi

**Reply by dai-shi** [reply]
Well, the inconsistencies are the issues when we use `get` on read.
So, in your use case it won't be a big issue. (but the library doesn't distinguish the difference.)
Anyway, writing directly to derived atoms is prohibited, because it's derived.

**Reply by tukaz** [reply]
this worked. i think this example has to be more visible in the documentation since is very common we get something async and we have to modify it later.
